core/store/orm/orm.go

```diff
@@ -326,6 +339,44 @@ func (orm *ORM) LinkEarnedFor(spec *models.JobSpec) (*assets.Link, error) {
 	return earned, nil
 }
 
+// UpsertErrorFor upserts a JobSpecError record, incrementing the occurences counter by 1
+// if the record is found
+func (orm *ORM) UpsertErrorFor(jobID *models.ID, description string) {
+	orm.MustEnsureAdvisoryLock()
+	err := orm.db.Exec(`
+		INSERT INTO job_spec_errors (
+			job_spec_id, description, created_at, updated_at
+		) VALUES (
+				?, ?, ?, ?
+		) ON CONFLICT (job_spec_id, description)
+		DO UPDATE
+		SET occurences = job_spec_errors.occurences + 1, updated_at = ?
+`, jobID, description, time.Now(), time.Now(), time.Now()).Error
+
+	if err != nil {
+		logger.Error(errors.Wrap(err, "Unable to create JobSpecError"))
+	}
+}
+
+// FindJobSpecError looks for a JobSpecError record with the given jobID and description
+func (orm *ORM) FindJobSpecError(jobID *models.ID, description string) (*models.JobSpecError, error) {
+	jobSpecErr := &models.JobSpecError{}
+	err := orm.db.
+		Where("job_spec_id = ? AND description = ?", jobID, description).
+		First(&jobSpecErr).Error
+	return jobSpecErr, err
+}
+
+// DeleteJobSpecError removes a JobSpecError
+func (orm *ORM) DeleteJobSpecError(ID uint) error {
+	orm.MustEnsureAdvisoryLock()
+	err := orm.db.Where("id = ?", ID).First(&models.JobSpecError{}).Error
+	if err != nil {
+		return err
+	}
+	return orm.db.Delete(&models.JobSpecError{ID: ID}).Error
```

Comment:
You can save a query here by simply doing

`DELETE FROM job_spec_errors WHERE id = ?`

Reply:
ahhh I figured there must be a way to do this. Thanks!

Reply:
This still doesn't seem to work - if the ID doesn't exist in the DB then delete still returns no error. I think we want a way to return a NotFound error instead.

Reply:
```go
// returns nil even if ID is not found
return orm.db.Exec("DELETE FROM job_spec_errors WHERE id = ?", ID).Error
```

Reply:
Not sure what the most gorm-ish way to handle this is. 2 queries seems unnecessary for something so simple, but I also don't want to write a confusing query just to perform a simple delete :sweat_smile:.

Reply:
Why bother returning a not found error? If it's not there, seems like it was "deleted" right? Either way, we got the desired outcome.

Reply:
I think this is a perfectly fine solution:

```
// returns nil even if ID is not found
return orm.db.Exec("DELETE FROM job_spec_errors WHERE id = ?", ID).Error
```

Reply:
I agree this seems okay just from the perspective of the ORM method. But this gets used in the controller to delete an error record. If we remove the check that the record exists in the first place (without other changes) then delete requests could succeed (return 200) without having actually done anything. I think it would be better to show error messages when our front end does something buggy like tries to send a delete request for a non-existent record. I could refactor the code path so that we do the check for existence separately, in the controller - and maybe that would have better consistency with other orm delete methods? But yeah point of the story - I'm worried about not showing errors messages when something buggy happens in the CLI or the GUI.